integrations/refined/src/main/scala/sttp/tapir/codec/refined/TapirCodecRefined.scala

```diff
@@ -11,6 +12,11 @@ import shapeless.Witness
 import scala.reflect.ClassTag
 
 trait TapirCodecRefined extends LowPriorityValidatorForPredicate {
+
+  implicit def refinedTapirStringSchema[P]: Schema[String Refined P] = Schema(SString)
+  implicit def refinedTapirIntSchema[P]: Schema[Int Refined P] = Schema(SInteger)
+  implicit def refinedTapirLongSchema[P]: Schema[Long Refined P] = Schema(SInteger)
```

Comment:
maybe a more generic version, sth like:

```
implicit def refinedTapirSchema[V, P](implicit vSchema: Schema[V]): Schema[V Refined P] = vSchema
```

would work?

Reply:
Maybe. They idea is to shortcut the compilers' implicit search. It could be that this doesn't make it short enough still.
But, I'll try it tomorrow to see if it works for my use cases.

Reply:
Right, but maybe we could make it work for other "raw" datatypes as well (without copying everything from `Schema`).

Another thing - could you maybe add a test that this works? Even a simple test that contains the endpoint (without checking anything) would be enough - the test here would be that the code compiles (and compilation would mean "test passed")

Reply:
In my code base, I replaced the 3 implicits with the following one, and then it still compiles fine.

```scala
implicit def refinedTapirSchema[V, P](implicit vSchema: Schema[V]): Schema[V Refined P] = Schema(vSchema.schemaType)
```

Returning `vSchema` doesn't work because it is of the wrong type. Should this return `Schema(vSchema.schemaType, vSchema.isOptional, vSchema.description, vSchema.format)` instead?

I'll try to create a unit test that fails without this line. I should be able to replicate that from our code base.

Reply:
There is `Schema.as` available if you import `sttp.tapir.internal._` - to be used only when you know what you're doing :)